core/job/service/job_service.go

```diff
@@ -333,18 +336,25 @@ func (j *JobService) Refresh(ctx context.Context, projectName tenant.ProjectName
 	return errors.MultiToError(me)
 }
 
-func (j *JobService) Validate(ctx context.Context, jobTenant tenant.Tenant, jobSpecs []*job.Spec, logWriter writer.LogWriter) error {
+func (j *JobService) Validate(ctx context.Context, jobTenant tenant.Tenant, jobSpecs []*job.Spec, jobNamesWithInvalidSpec []job.Name, logWriter writer.LogWriter) error {
 	me := errors.NewMultiError("validate specs errors")
 
 	tenantWithDetails, err := j.tenantDetailsGetter.GetDetails(ctx, jobTenant)
 	if err != nil {
 		return err
 	}
 
-	jobs, err := j.generateJobs(ctx, tenantWithDetails, jobSpecs, logWriter)
+	existingJobs, err := j.repo.GetAllByTenant(ctx, jobTenant)
+	me.Append(err)
+
+	toAdd, toUpdate, toDelete, err := j.differentiateSpecs(existingJobs, jobTenant, jobSpecs, jobNamesWithInvalidSpec)
```

Comment:
can we return the unmodified job on this function? then we can simplify [getAllJobsToValidateMap](https://github.com/goto/optimus/pull/31/files#diff-390a4347f57303777e426d01a409c4e8664680983672226c51044a189280a849R403) to just only passed the incoming and unmodified

Reply:
sure, maybe I will just return the unmodified specs (not unmodified jobs) since it is in differentiate specs function and handle it in the getAllJobsToValidateMap